src/components/AuthForm.tsx

```diff
@@ -0,0 +1,238 @@
+      } else if (error.message === "auth/email-already-in-use") {
+        setErrorMessage(ERROR_MESSAGES.EMAIL_IN_USE);
+      }
+    }
+  };
+
+  const isFormValid = () => {
+    if (isSignIn) {
+      return authData.email && authData.password && errors.length === 0;
+    } else {
+      return (
+        authData.email &&
+        authData.password &&
+        authData.username &&
+        authData.confirmPassword &&
+        errors.length === 0
+      );
+    }
+  };
+
+  const renderErrorMessage = (fieldName: string, message: string) => {
+    return (
+      <div key={fieldName} className={classes["form-input__error"]}>
+        {message}
+      </div>
+    );
+  };
+
+  const renderFieldErrorMessage = (fieldName: string) => {
+    const error = errors.find((error) => error.fieldName === fieldName);
+    if (error) {
+      return renderErrorMessage(fieldName, error.message);
+    }
+    return null;
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className={classes.form}>
+      {isSuccess && (
+        <div className={classes["success-message"]}>Successful sign up!</div>
+      )}
+      {!isSignIn && (
+        <div>
+          <input
+            type="text"
+            id="username"
+            name="username"
+            placeholder="Username"
+            value={authData.username || ""}
+            onChange={handleInputChange}
+            onBlur={handleBlur}
+            className={classes["form-input"]}
+          />
+          {renderFieldErrorMessage("username")}
+        </div>
+      )}
+      <div>
+        <input
+          type="email"
+          id="email"
+          name="email"
+          placeholder="Email"
+          value={authData.email}
+          onChange={handleInputChange}
+          onBlur={handleBlur}
+          className={classes["form-input"]}
+        />
+        {renderFieldErrorMessage("email")}
+      </div>
+      <div>
+        <input
+          type="password"
+          id="password"
+          name="password"
+          placeholder="Password"
+          value={authData.password}
+          onChange={handleInputChange}
+          onBlur={handleBlur}
+          className={classes["form-input"]}
+        />
+        {renderFieldErrorMessage("password")}
+      </div>
+      <div>
+        {!isSignIn && (
+          <div>
+            <input
+              type="password"
+              id="confirmPassword"
+              name="confirmPassword"
+              placeholder="Confirm Password"
+              value={authData.confirmPassword || ""}
+              onChange={handleInputChange}
+              onBlur={handleBlur}
+              className={classes["form-input"]}
+            />
+            {renderFieldErrorMessage("confirmPassword")}
+          </div>
+        )}
+      </div>
+      <div className={classes["btn-container"]}>
+        <Button type="submit" variant="secondary" disabled={!isFormValid()}>
+          {isSignIn ? "Sign In" : "Sign Up"}
```

Comment:
Minor, but lets strive for perfection :D Move text to constants in a global text file